server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -164,7 +164,7 @@ protected StarTreeDocument getStarTreeDocument(
             // actual indexing field they're based on
             metrics[i] = metricAggregatorInfos.get(i)
                 .getValueAggregators()
-                .toStarTreeNumericTypeValue(metricDocValuesIterator.value(currentDocId));
+                .toAggregatedValueType(metricDocValuesIterator.value(currentDocId));
```

Comment:
The abstract builder tests didn't catch this as all fields were of double type right ? should we change any fields value to a different type like long/int ?